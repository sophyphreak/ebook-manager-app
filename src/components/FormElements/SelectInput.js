import React, { Component } from 'react';
import {
  FormGroup,
  Label,
  Input
} from 'reactstrap';

const SelectInput = ({ label, options, value, onChange }) => (
  <FormGroup>
    {label && <Label>{label}</Label>}
    <Input
      type="select"
      onChange={onChange}
    >
      {
        options.map((option, i) => (
          <option
            value={option}
            key={i}
          >
            {option}
          </option>
        ))
      }
    </Input>
  </FormGroup>
);

export default SelectInput;  