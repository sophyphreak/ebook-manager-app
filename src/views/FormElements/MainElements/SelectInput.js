import React, { Component } from 'react';
import { FormFeedback, FormGroup, Label, Input } from 'reactstrap';
import getValidity from '../getValidity/getValidity';

const SelectInput = ({
  hasError,
  errorMessage,
  label,
  options,
  value,
  onChange
}) => (
  <FormGroup>
    {label && <Label>{label}</Label>}
    <Input
      type="select"
      onChange={onChange}
      valid={getValidity(hasError, errorMessage)}
    >
      {options.map((option, i) => (
        <option value={option} key={i}>
          {option}
        </option>
      ))}
    </Input>
    {errorMessage && <FormFeedback>{errorMessage}</FormFeedback>}
  </FormGroup>
);

export default SelectInput;
