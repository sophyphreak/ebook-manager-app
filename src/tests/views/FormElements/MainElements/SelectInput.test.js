import React from 'react';
import { shallow } from 'enzyme';
import SelectInput from '../../../../views/FormElements/MainElements/SelectInput';

let options, wrapper;

beforeEach(() => {
  options = ['option 0', 'option 1', 'option 2'];
  wrapper = shallow(<SelectInput options={options} />);
});

test('should render SelectInput correctly', () => {
  expect(wrapper).toMatchSnapshot();
});
