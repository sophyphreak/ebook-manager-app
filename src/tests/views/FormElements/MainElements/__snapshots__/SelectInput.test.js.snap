// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render SelectInput correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <SelectInput
    options={
      Array [
        "option 0",
        "option 1",
        "option 2",
      ]
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": Array [
        undefined,
        <Input
          invalid={false}
          type="select"
          valid={false}
        >
          <option
            value="option 0"
          >
            option 0
          </option>
          <option
            value="option 1"
          >
            option 1
          </option>
          <option
            value="option 2"
          >
            option 2
          </option>
        </Input>,
        undefined,
      ],
      "tag": "div",
    },
    "ref": null,
    "rendered": Array [
      undefined,
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": Array [
            <option
              value="option 0"
            >
              option 0
            </option>,
            <option
              value="option 1"
            >
              option 1
            </option>,
            <option
              value="option 2"
            >
              option 2
            </option>,
          ],
          "invalid": false,
          "onChange": undefined,
          "type": "select",
          "valid": false,
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": "0",
            "nodeType": "host",
            "props": Object {
              "children": "option 0",
              "value": "option 0",
            },
            "ref": null,
            "rendered": "option 0",
            "type": "option",
          },
          Object {
            "instance": null,
            "key": "1",
            "nodeType": "host",
            "props": Object {
              "children": "option 1",
              "value": "option 1",
            },
            "ref": null,
            "rendered": "option 1",
            "type": "option",
          },
          Object {
            "instance": null,
            "key": "2",
            "nodeType": "host",
            "props": Object {
              "children": "option 2",
              "value": "option 2",
            },
            "ref": null,
            "rendered": "option 2",
            "type": "option",
          },
        ],
        "type": [Function],
      },
      undefined,
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": Array [
          undefined,
          <Input
            invalid={false}
            type="select"
            valid={false}
          >
            <option
              value="option 0"
            >
              option 0
            </option>
            <option
              value="option 1"
            >
              option 1
            </option>
            <option
              value="option 2"
            >
              option 2
            </option>
          </Input>,
          undefined,
        ],
        "tag": "div",
      },
      "ref": null,
      "rendered": Array [
        undefined,
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": Array [
              <option
                value="option 0"
              >
                option 0
              </option>,
              <option
                value="option 1"
              >
                option 1
              </option>,
              <option
                value="option 2"
              >
                option 2
              </option>,
            ],
            "invalid": false,
            "onChange": undefined,
            "type": "select",
            "valid": false,
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": "0",
              "nodeType": "host",
              "props": Object {
                "children": "option 0",
                "value": "option 0",
              },
              "ref": null,
              "rendered": "option 0",
              "type": "option",
            },
            Object {
              "instance": null,
              "key": "1",
              "nodeType": "host",
              "props": Object {
                "children": "option 1",
                "value": "option 1",
              },
              "ref": null,
              "rendered": "option 1",
              "type": "option",
            },
            Object {
              "instance": null,
              "key": "2",
              "nodeType": "host",
              "props": Object {
                "children": "option 2",
                "value": "option 2",
              },
              "ref": null,
              "rendered": "option 2",
              "type": "option",
            },
          ],
          "type": [Function],
        },
        undefined,
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
